spec/decorators/application_choice_export_decorator_spec.rb

```diff
@@ -111,4 +111,41 @@
       expect(result).to eq(%w[GB IE US])
     end
   end
+
+  describe 'nationality' do
+    let(:decorated_application_form) { described_class.new(application_choice) }
+    let(:application_choice) { create(:application_choice, application_form: application_form) }
+
+    context 'when dual nationality including British' do
+      let(:application_form) { create(:application_form, first_nationality: 'Cypriot', second_nationality: 'British') }
+
+      it 'they should be coded as GB' do
+        expect(decorated_application_form.nationality).to eq('GB')
+      end
+    end
+
+    context 'when dual nationality, not including British, but including non-UK EU country' do
+      let(:application_form) { create(:application_form, first_nationality: 'Cypriot', second_nationality: 'American') }
+
+      it 'they should be coded as the EU country code' do
+        expect(decorated_application_form.nationality).to eq('CY')
+      end
+    end
+
+    context 'when dual nationality and both are non-UK EU countries' do
+      let(:application_form) { create(:application_form, first_nationality: 'Greek', second_nationality: 'Hungarian') }
+
+      it 'they should be coded as the EU country code' do
+        expect(decorated_application_form.nationality).to eq('GR')
+      end
+    end
+
+    context 'when dual nationality and both are neither is British or EU' do
+      let(:application_form) { create(:application_form, first_nationality: 'Indian', second_nationality: 'American') }
+
+      it 'they can be coded as either country' do
```

Comment:
```suggestion
      it 'they are coded as the first country' do
```

Reply:
The condition that we need to satisfy is that they can be either country code, I just decided to return the first one as it seemed a bit pointless shuffling the data just to get a random result in this instance.

Reply:
I guess you're stating what our business rule is, as a form of documentation, rather than focusing on the implementation? I think there is some value in preserving the nationality order specified by the candidate (which we do, unless we promote any of their other nationalities to assist them with funding), and being explicit about it in the specs. It's not a big thing, but saying 'either' makes me expect we're shuffling the array in the code, which we don't do.

Reply:
👍🏼 Reworded as part of moving to a new decorator